DSharpPlus/Clients/DiscordClient.Dispatch.cs

```diff
@@ -1705,9 +1705,12 @@ internal async Task OnVoiceStateUpdateEventAsync(JObject raw)
             var vstateNew = raw.ToObject<DiscordVoiceState>();
             vstateNew.Discord = this;
 
-            gld._voiceStates.TryGetValue(uid, out var vstateOld);
+            gld._voiceStates.TryRemove(uid, out var vstateOld);
 
-            gld._voiceStates[vstateNew.UserId] = vstateNew;
+            if (vstateNew.Channel is not null)
```

Comment:
Rather than always removing it, you can probably just check if it's not null and then assign the voice state to the cache, and otherwise if it's null then remove it. Like so: 

```csharp
if(vstateNew.Channel != null)
     // assign to cache...
else
    // remove from cache...
```

Reply:
Since `TryRemove()` returns the old voice state I figured I could avoid using a discard, i.e.
```cs
gld._voiceStates.TryGetValue(uid, out var vstateOld);

if (vstateNew.Channel is not null)
{
    gld._voiceStates[vstateNew.UserId] = vstateNew;
}
else
{
    gld._voiceStates.TryRemove(uid, out _);
}
```

Reply:
Fair.